template/osfamilymap.yaml

```diff
@@ -40,4 +40,9 @@ Solaris: {}
 
 Windows: {}
 
-MacOS: {}
+  {% if grains.os == 'MacOS' %}
+MacOS:
+  rootgroup: {{ salt['cmd.run']("stat -f '%Sg' /dev/console") }}
+  service:
+    group: {{ salt['cmd.run']("stat -f '%Sg' /dev/console") }}
+  {% endif %}
```

Comment:
So there's no standard `rootgroup` and it has to come via. that `cmd.run`?  If that's the only way, then do you mind if we tidy this up a little and remove the dupe?  How about the following suggestions:

We do the "heavy-lifting" Jinja at the top of the file:

```sls
---
{%- if grains.os == 'MacOS' %}
{%-   set macos_group = salt['cmd.run']("stat -f '%Sg' /dev/console") %}
{%- endif %}
```

And then use it below like so:

```sls
MacOS:
  rootgroup: {{ macos_group }}
  service:
    group: {{ macos_group }}
```

Reply:
Sure that style works for me. I will try to remember to raise a salt issue asking for `sysuser` and `sysgroup` grains to be made available on MacOS as this is common issue.

Still I need to wrap usage in jinja to avoid a jinja rendering error on `macos_group`.

Reply:
Should `service:group` be added to defaults.yaml too? See grafana-formula.

Reply:
> Still I need to wrap usage in jinja to avoid a jinja rendering error on `macos_group`.

That's because you changed the variable names from `macos_group`!  There's a reason to avoid wrapping this section in a Jinja `if`: there's a way of simulating `map.jinja` for different OSes, but with this `if` it can't be done.  Basically, using the changes I suggested at the top of the file, you can remove the `if` and `endif` Jinja blocks now.

> Should `service:group` be added to defaults.yaml too? See grafana-formula.

Let's do it when there's a need for it.  If the need is there now, please feel free.

Reply:
thanks for review @myii   You are correct.  I have updated the commit accordingly.

So on non-MacOS the `rootgroup` variable (at bottom of osfamilymap.yaml) will resolve?  I was'nt aware of that.